Splitio.Redis/Services/Cache/Classes/RedisCacheBase.cs

```diff
@@ -10,33 +14,45 @@ public abstract class RedisCacheBase
         protected string MachineIp;
         protected string MachineName;
 
-        public RedisCacheBase(string userPrefix = null)
+        protected RedisCacheBase(RedisConfig redisConfig = null, bool clusterMode = false)
         {
-            UserPrefix = userPrefix;
+            UserPrefix = redisConfig.RedisUserPrefix;
             RedisKeyPrefix = "SPLITIO.";
-
-            if (!string.IsNullOrEmpty(userPrefix))
-            {
-                RedisKeyPrefix = userPrefix + "." + RedisKeyPrefix;
-            }
+            UpdateRedisKeyPrefix(UserPrefix, redisConfig, clusterMode);
         }
 
         public RedisCacheBase(string machineIP,
             string sdkVersion,
             string machineName,
-            string userPrefix = null)
+            RedisConfig redisConfig = null, bool clusterMode = false)
```

Comment:
yo can remove `machineIP`, `sdkVersion` and `machineName` because you are passing `RedisConfig` object now.